modular_skyrat/modules/salon/code/pattern_table.dm

```diff
@@ -0,0 +1,167 @@
+			printing = TRUE
+			if(do_after(usr, 3 SECONDS, src))
+				var/obj/item/pattern_kit/new_pattern = new(get_turf(src))
+				new_pattern.clothing_datum = SSclothing_database.clothing_loaded[selected_pattern]
+				new_pattern.update_details()
+			printing = FALSE
+			. = TRUE
+		if("ban_pattern_admin")
+			var/is_admin = is_admin(usr.client)
+			if(!is_admin)
+				balloon_alert(usr, "not an admin!") // The button shouldn't be available, but cheeky fuckers messing with tgui could send this input anyways so we sanity check.
+				return
+			var/selected_pattern = params["pattern_id"]
+			var/are_you_sure = tgui_alert(usr, "ADMIN: Are you sure you want to ban this pattern?", "Pattern Ban", list("Yes", "No"))
+			if(are_you_sure == "Yes")
+				var/datum/tailor_clothing/clothing_datum = SSclothing_database.clothing_loaded[selected_pattern]
+				clothing_datum.ban_pattern()
+				message_admins("[key_name(usr)] banned [clothing_datum.name] (ID #[clothing_datum.id]) from the pattern database.")
+				log_admin("[key_name(usr)] banned [clothing_datum.name] (ID #[clothing_datum.id]) from the pattern database.")
+			. = TRUE
+
+/obj/machinery/pattern_table/ui_data(mob/user)
+	var/list/data = list()
+	var/list/patterns = list()
+	var/is_admin = is_admin(user.client)
+	for(var/datum/tailor_clothing/clothing_datum in SSclothing_database.clothing_loaded)
+		if(selected_category == clothing_datum.slot && !clothing_datum.banned)
+			patterns.Add(list(clothing_datum.get_list()))
+			var/icon/inventory_icon = new(file("data/clothing_icons/[clothing_datum.id].dmi"), "inventory")
+			user << browse_rsc(inventory_icon, "clothing_[clothing_datum.id].png")
+	data["patterns"] = patterns
+	data["category"] = selected_category
+	data["is_admin"] = is_admin
+	return data
+
+/obj/machinery/pattern_table/attack_hand(mob/living/user, list/modifiers)
+	. = ..()
+	if(LAZYACCESS(modifiers, RIGHT_CLICK))
+		print_patterns(user, modifiers)
+	else
+		if(is_banned_from(user.ckey, BAN_CLOTHING))
+			to_chat(user, "You are banned from uploading custom clothing.")
+			return
+		create_pattern(user, modifiers)
+
+/obj/machinery/pattern_table/proc/print_patterns(mob/living/user, list/modifiers)
+	ui_interact(user)
+	return
+
+/obj/machinery/pattern_table/proc/create_pattern(mob/living/user, list/modifiers)
+	var/has_dmi = tgui_alert(user, "Do you have a prepared .dmi file?", "Pattern Design", list("Yes", "No"))
+	if(!has_dmi || has_dmi == "No")
+		user.balloon_alert(user, "see chat!")
+		to_chat(user, "Please see [DMI_TUTORIAL_URL] for an example of a properly set up 32x32 .dmi file from inside Dream Maker.")
+		to_chat(user, "If the URL does not function for you, your .dmi file will need an onmob icon state with the 4 cardinal directions, and an inventory icon state.")
+		to_chat(user, "If you want your clothing to support digitigrade, you can add an onmob_digit icon state. This is optional, and isn't required.")
+		to_chat(user, "Clothing .dmi files must be 32x32.")
```

Comment:
```suggestion
		to_chat(user, span_warning("Please see [DMI_TUTORIAL_URL] for an example of a properly set up 32x32 .dmi file from inside Dream Maker."))
		to_chat(user, span_warning("If the URL does not function for you, your .dmi file will need an 'onmob' icon state with the 4 cardinal directions, and an 'inventory' icon state."))
		to_chat(user, span_warning("If you want your clothing to support digitigrade, you can add an 'onmob_digit' icon state. This is optional, and isn't required."))
		to_chat(user, span_warning("Clothing .dmi files must be 32x32."))
```
Catches the eye more with the warning(and doesnt appear as plain text), and the 'quotes' make it easier to tell thats the icon state's NAME and not just the description